docs/shared/reference/nx-json.md

```diff
@@ -22,10 +22,9 @@ The following is an expanded example showing all options. Your `nx.json` will li
     "production": ["!{projectRoot}/**/*.spec.tsx"]
   },
   "targetDefaults": {
-    "build": {
+    "@nx/js:tsc": {
```

Comment:
tbh I never used this syntax yet. So this updates all targets using the `@nx/js:tsc` executor. Is there a way to define the target name in this config as well? So have both?

I also think it's still valuable to have the alternative version of defining just the target name like `build`, `lint` etc and then have another one that shows how to "target" a specific executor. There is one such example further down in the `Outputs` section. Maybe we could just merge those and use a similar one.

Reply:
> tbh I never used this syntax yet. So this updates all targets using the `@nx/js:tsc` executor. Is there a way to define the target name in this config as well? So have both?

It will apply to all targets using that executor. It's explained here https://nx-dev-git-fork-leosvelperez-docs-update-targets-defaults-nrwl.vercel.app/reference/nx-json#target-defaults. 

You can have both, but the `targetDefaults` for an executor will override the `targetDefaults` for a target that uses the executor. They are not merged and the executor one takes precedence.

> I also think it's still valuable to have the alternative version of defining just the target name like `build`, `lint` etc and then have another one that shows how to "target" a specific executor. There is one such example further down in the `Outputs` section. Maybe we could just merge those and use a similar one.

I'm not sure we need to duplicate examples with that. We'd need to do it everywhere. We've always supported both, and most of the time we were only suggesting using target names as keys. Now with Project Crystal, we generate `targetDefaults` using executors as the key. So, we're replacing some relevant examples to show that a bit more. We're not hiding you can use targets as keys. In the link I shared above, which is in this same document, it's explained that you can use both, and as you mentioned, throughout our docs there will be examples of both. I'm not replacing all examples with executors.

Reply:
Sure, I'm not saying duplicate across the board, but ppl are still familiar seeing the `targetName` config approach, in particular when it comes to defining caching related config or `dependsOn` as it is usually related to a target.
That's why I mentioned having both options as someone using a npm workspace repo looking at this would be really confused because they wouldn't even use Nx plugins etc. While someone using Nx Crystal plugins might want to see your setup there.

Reply:
That's fine. I still don't think it's too relevant given that most examples in this same page and in all the docs mostly use target names anyways (so users not using executors have plenty of examples showing what they have), but I also think it doesn't hurt to add it, so I added a separate target default using a target to this particular snippet 👍🏻 

This snippet is only an example showcasing the structure of an `nx.json` file. Users don't stop at this snippet and leave with all the information they need. When users go to a reference page, they look for definitions, and the definition for `targetDefaults` explains that you can use an executor or a target name as the key. It describes how they are applied, what the precedence is, and a few examples of using one or the other. So, IMHO, anyone coming to the page will still read the description of the options they're interested in, and that's what needs to show and explain anything needed.